builder/azure/common/template/template_builder.go

```diff
@@ -533,7 +533,7 @@ func (s *TemplateBuilder) SetSecurityProfile(secureBootEnabled bool, vtpmEnabled
 		resource.Properties.SecurityProfile.UefiSettings.SecureBootEnabled = common.BoolPtr(secureBootEnabled)
 		resource.Properties.SecurityProfile.UefiSettings.VTpmEnabled = common.BoolPtr(vtpmEnabled)
 	}
-	resource.Properties.SecurityProfile.EncryptionAtHost = common.BoolPtr(encryptionAtHost)
+	resource.Properties.SecurityProfile.EncryptionAtHost = encryptionAtHost
```

Comment:
I wonder, should we only set the parameter if true instead of always setting a value? This would be a different (simpler) implementation than what you propose here, that being said maybe there's a case for setting `encryptionAtHost` to `false` manually?
If so the current approach makes sense and we can accept it as-is.

cc @JenGoldstrich since you know Azure much better than I do

Reply:
I'm conflicted tbh, to do what you suggest it would need to be refactored to be something like this
```
if encryptionAtHost != nil && *encryptionAtHost {
     	resource.Properties.SecurityProfile.EncryptionAtHost = encryptionAtHost
}
```

While we would avoid setting the the field in the template by changing to this, it requires that extra nil check and a true/false. check, when we never actually send false, only true and nil, since sending false is what causes this issue for Azure Subscriptions without this feature enabled (TBH this feels like an API bug but thats an aside). But I will accept either solution tbh